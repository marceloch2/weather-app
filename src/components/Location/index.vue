<template>
  <form class="location" ref="locationForm">
    <fieldset>
      <legend>{{ title.toUpperCase() }}</legend>

      <label for="today__city">City Name</label>
      <input
        autofocus
        type="text"
        id="today__city"
        class="today__city"
        placeholder="city name"
        v-model.trim="city"
      >

      <label v-if="hideCountry" for="today__country">Country Code</label>
      <input
        v-if="hideCountry"
        type="text"
        id="today__country"
        class="today__country"
        placeholder="country code"
        v-model.trim="country"
      >
      <button type="submit">Update Data</button>
    </fieldset>
  </form>
</template>

<script>
import Location from "./location";
export default Location;
</script>

<style lang="sass" scoped>
	@import 'location.scss'
</style>
