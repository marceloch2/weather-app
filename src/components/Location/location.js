import { mapState } from 'vuex'

import weatherApiService from '../../Services/weatherApiService'

export default {
  name: 'today',
  data() {
    return {
      title: 'Choose your location',
      city: '',
      country: ''
    }
  },

  mounted() {},

  computed: mapState({
    location: state => state.location,
    tab: state => state.tab
  }),

  watch: {
    tab: function(_old, _new) {
      if (this.location.city !== '') {
        this.search()
      }
    }
  },

  methods: {
    async search() {
      let tab = this.$store.state.tab

      this.$store.commit('setCurrentLocation', {
        city: this.city,
        country: this.country
      })

      try {
        let weatherData = await weatherApiService.fetchWeather(
          this.$store.state.location.city,
          this.$store.state.location.country,
          tab
        )
        this.$store.commit('setWeatherData', {
          weatherData,
          tab
        })
      } catch (error) {
        this.$store.commit(
          'globalError',
          `No Weather data found for ${this.$store.state.location.city}`
        )
      }
    }
  }
}
