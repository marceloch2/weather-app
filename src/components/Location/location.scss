.location {
  width: 400px;
  text-align: left;
  padding: 10px;

  fieldset {
    border: none;
    border-top: solid 1px darkslategray;
    padding: 30px;
    font-weight: bold;
  }

  input {
    width: 80%;
    padding: 10px;
    margin-bottom: 15px;
    border: solid 1px darkslategray;
  }

  button {
    padding: 10px;
    width: 50%;
    border: none;
    background-color: darkslategray;
    color: #fff;
  }
}
