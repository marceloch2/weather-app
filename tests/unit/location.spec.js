import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils'
import Location from '@/components/Location'
import weatherData from './mocks/today'

import createStateConfig from './createStateConfig'

const localVue = createLocalVue()
localVue.use(Vuex)

const stateConfig = createStateConfig()

const store = new Vuex.Store({
  state: stateConfig,
  mutations: {
    setCurrentLocation: jest.fn(),
    globalError: jest.fn(),
    setWeatherData: jest.fn(),
    setIsLoading: jest.fn(),
    fetchWeather: jest.fn()
  }
})

const $router = {
  push: jest.fn()
}

const wrapper = shallowMount(Location, {
  store,
  localVue,
  mocks: {
    $router
  }
})

describe('Location.vue', () => {
  beforeEach(() => {})

  it('Snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Mount with initial state', () => {
    expect(wrapper.vm.title).toMatch('Choose your location')
    expect(wrapper.vm.city).toBe('')
    expect(wrapper.vm.country).toBe('')
    expect(wrapper.vm.hideCountry).toBe(false)
  })

  it('Can call Search', async () => {
    wrapper.vm.city = 'Haarlem'
    wrapper.vm.$store.state.isLoading = false
    wrapper.vm.$store.state.weatherData_weather = weatherData
    wrapper.vm.$store.state.location.city = 'Haarlem'

    expect(wrapper.vm.search()).toBeInstanceOf(Promise)
  })
})
