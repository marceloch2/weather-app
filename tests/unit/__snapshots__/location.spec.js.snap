// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location.vue Snapshot 1`] = `
<form class="location">
  <fieldset>
    <legend>CHOOSE YOUR LOCATION</legend> <label for="today__city">City Name</label> <input autofocus="autofocus" type="text" id="today__city" placeholder="city name" class="today__city">
    <!---->
    <!----> <button type="submit">Update Data</button>
  </fieldset>
</form>
`;
